modules/spring-config/src/main/java/org/mule/config/spring/SpringConfigurationBuilder.java

```diff
@@ -58,7 +59,10 @@ protected void doConfigure(MuleContext muleContext) throws Exception
         // Note: The SpringRegistry must be created before applicationContext.refresh() gets called because
         // some beans may try to look up other beans via the Registry during preInstantiateSingletons().
         muleContext.addRegistry(registry);
-        registry.initialise();
+        if (muleContext.getLifecycleManager().isPhaseComplete(Initialisable.PHASE_NAME))
```

Comment:
Why is the configuration builder doing lifecycle?  This should be in addRegisty() IMO.  Also once addRegisty() is gone, wondering what happending is you use multiple SpringConfigurationBuilders? How does this work, because each ConfigurationBuilder is creating and adding a new registry?  If it doesn't make sense to use mutiple ConfigurationBuilders any more, we need to give this some thought and work out if there are any API changes that can be made to reflect this.  The original ConfigruationBuilder interface was put together like this to specifically support/allow multiple configuration sources.